packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,157 @@
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * A single tag property values
+ */
+interface ITagProps {
+    value: string | Token;
+    propagate?: boolean;
+}
+
+/**
+ * A an object of tags
+ */
+export interface ITags {
+    [key: string]: ITagProps;
+}
+
+/**
+ * A single tag
+ */
+export interface ITag {
```

Comment:
I created this because I had to export an array of objects and this is the array. However, for `addTag` I use this interface which we could switch to `ITags` and we could support multiple tags (rename to `addTags`).  If we remove this from the `addTag` method then we can also drop propagate, and this really is `CloudFormationTag` if we capitalize `Key` and `Value`. I kept flip-flopping here and could use another opinion.

Reply:
I changed this design and use `ITags` with the renamed `addTags` and I eliminated the `propagate` member.

Reply:
I don't think this is needed

Reply:
I did use the `cloudformation.Tag` interface, I think that is a better fit. If you meant I should not have a reference to this data structure at all, then let me know.